Terminal.Gui/Core/MainLoop.cs

```diff
@@ -162,7 +162,9 @@ public bool RemoveTimeout (object token)
 			var idx = timeouts.IndexOfValue (token as Timeout);
 			if (idx == -1)
 				return false;
-			timeouts.RemoveAt (idx);
+			lock (timeouts) {
```

Comment:
Should this lock also cover the `IndexOfValue` call? If multithreading is a possibility.

Reply:
I was not already giving failing but I think you're right. Done.